### ISSUE TITLE
Can't get past share dialog on iOS

### ISSUE THREAD

**Issue by schoudha** [KasemJaffer/receive_sharing_intent#33]
Things work fine on Android but on iOS I get the share dialog and then it does nothing.

Any ideas?

**Comment by VictorGa** [comment]
Same here, I can't figure out why I can't pass the dialog in iOS. Check all the xcode configuration multiple times and getting always the same results.

Is there anybody who managed to have it working in iOS?

**Comment by nfodor** [comment]
Same here, does not go past the share screen on iOS.

**Comment by PabloCrucitta** [comment]
It happens the same to me.

**Comment by wrbl606** [comment]
Try the solution from https://github.com/KasemJaffer/receive_sharing_intent/issues/36 and tell if it helped.

**Comment by acoutts** [comment]
> Try the solution from [#36 (comment)](https://github.com/KasemJaffer/receive_sharing_intent/issues/36#issuecomment-595130324) and tell if it helped.

That fixed it for me!

**Comment by GRBurst** [comment]
@schoudha @VictorGa @PabloCrucitta @nfodor 
is this still an open issue?

**Comment by nfodor** [comment]
Yeah so on android no problem but ios i would like to avoid the dialog to write comment before sending, because after i tap send it works but the app i send it from freezes. 

On Apr 21, 2020, at 3:24 AM, GRBurst <notifications@github.com> wrote:

﻿


@schoudha @VictorGa @PabloCrucitta @nfodor
is this still an open issue?

**Comment by PabloCrucitta** [comment]
> @schoudha @VictorGa @PabloCrucitta @nfodor
> is this still an open issue?

Fixed, thank you very much!

**Comment by woutervanwijk** [comment]
> Yeah so on android no problem but ios i would like to avoid the dialog to write comment before sending, because after i tap send it works but the app i send it from freezes.  On Apr 21, 2020, at 3:24 AM, GRBurst <notifications@github.com> wrote: ﻿ @schoudha @VictorGa @PabloCrucitta @nfodor is this still an open issue? — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub, or unsubscribe.

I get this dialog also. But only on the first share action. The app btw is nog frozen, there seems to be a semi-transparent window which you can slide down

**Comment by woutervanwijk** [comment]
Just to inform: I got rid of the Post dialog by commenting out this in the info.plist: 
		
In hindsight quite obvious if you know more about the ios extension system

**Comment by woutervanwijk** [comment]
I got rid of the dialog by adding these two lines to the end of the redirectToHostApp function in ShareViewController.swift:
```
...
        extensionContext!.completeRequest(returningItems: [], completionHandler: nil)
        // these two:
        extensionContext!.completeRequest(returningItems: nil, completionHandler: nil)
        extensionContext!.cancelRequest(withError:NSError())
    }
```

**Comment by thegrxp** [comment]
> I got rid of the dialog by adding these two lines to the end of the redirectToHostApp function in ShareViewController.swift:
> 
> ```
> ...
>         extensionContext!.completeRequest(returningItems: [], completionHandler: nil)
>         // these two:
>         extensionContext!.completeRequest(returningItems: nil, completionHandler: nil)
>         extensionContext!.cancelRequest(withError:NSError())
>     }
> ...
> ```

Thanks it's working

**Comment by ThuyenPVT** [comment]
> I got rid of the dialog by adding these two lines to the end of the redirectToHostApp function in ShareViewController.swift:
> 
> ```
> ...
>         extensionContext!.completeRequest(returningItems: [], completionHandler: nil)
>         // these two:
>         extensionContext!.completeRequest(returningItems: nil, completionHandler: nil)
>         extensionContext!.cancelRequest(withError:NSError())
>     }
> ...
> ```

thank you so much ! it's working, but i cannot receive media file, text on ios device
